packages/driver/cypress/integration/commands/actions/attachFile_spec.js

```diff
@@ -0,0 +1,673 @@
+        expect(input[0].files[0].type).to.eq('')
+        expect(input[0].files[0].lastModified).to.be.closeTo(Date.now(), 1000)
+      })
+
+      cy.get('#basic')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eql('foo')
+      })
+    })
+
+    it('attaches multiple files', () => {
+      cy.get('#multiple')
+      .attachFile([
+        {
+          contents: '@foo',
+          fileName: 'foo.txt',
+        }, {
+          contents: Buffer.from('{"a":"bar"}'),
+          fileName: 'bar.json',
+        },
+        Buffer.from('baz'),
+      ])
+
+      cy.get('#multiple')
+      .should('include.value', 'foo.txt')
+      .then((input) => {
+        expect(input[0].files[0].name).to.eq('foo.txt')
+        expect(input[0].files[1].name).to.eq('bar.json')
+        expect(input[0].files[2].name).to.eq('')
+      })
+
+      cy.get('#multiple')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eq('foo')
+        expect(contents[1]).to.eq('{"a":"bar"}')
+        expect(contents[2]).to.eq('baz')
+      })
+    })
+
+    it('attaches files with custom lastModified', () => {
+      cy.get('#basic').attachFile({
+        contents: '@foo',
+        lastModified: 1234,
+      })
+
+      cy.get('#basic').then((input) => {
+        expect(input[0].files[0].lastModified).to.eq(1234)
+      })
+    })
+
+    it('attaches files to an input from a label', () => {
+      cy.get('#basic-label').attachFile({ contents: '@foo' })
+
+      cy.get('#basic')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eql('foo')
+      })
+    })
+
+    it('attaches files to an input from a containing label', () => {
+      cy.get('#containing-label').attachFile({ contents: '@foo' })
+
+      cy.get('#contained')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eql('foo')
+      })
+    })
+
+    it('invokes change and input events on the input', (done) => {
+      const $input = cy.$$('#basic')
+
+      $input.on('input', (e) => {
+        const obj = _.pick(e.originalEvent, 'bubbles', 'cancelable', 'composed', 'target', 'type')
+
+        expect(obj).to.deep.eq({
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          target: $input.get(0),
+          type: 'input',
+        })
+
+        $input.on('change', (e) => {
+          const obj = _.pick(e.originalEvent, 'bubbles', 'cancelable', 'composed', 'target', 'type')
+
+          expect(obj).to.deep.eq({
+            bubbles: true,
+            cancelable: false,
+            composed: false,
+            target: $input.get(0),
+            type: 'change',
+          })
+
+          done()
+        })
+      })
+
+      cy.get('#basic').attachFile({ contents: '@foo' })
+    })
+
+    it('bubbles events', (done) => {
+      cy.window().then((win) => {
+        $(win).on('input', () => {
+          done()
```

Comment:
Since we already have a test asserting that the events are emitted directly on the targeted element, I don't think it's necessary to add an assertion here (it'd just be testing that the browser bubbles events properly, rather than any code we wrote).